### ISSUE TITLE
Support for Azure File Storage performance tiers

### ISSUE THREAD

**Issue by BHoggs** [hashicorp/terraform-provider-azurerm#8716]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure file shares support four performance tiers: 
* Premium
* Transaction Optimized
* Hot
* Cool

Currently have to change this manually after the share is created by terraform

### New or Affected Resource(s)

* azurerm_storage_share

### Potential Terraform Configuration

```hcl
resource "azurerm_storage_share" "file_share" {
    name                 = "MyFileShare"
    storage_account_name = azurerm_storage_account.example
    quota                = 5120
    tier                 = "Hot"
}
```

### References

* https://docs.microsoft.com/en-gb/azure/storage/files/storage-files-planning?WT.mc_id=Portal-Microsoft_Azure_FileStorage#storage-tiers

**Comment by nexxai** [comment]
Submitted PR [here](https://github.com/terraform-providers/terraform-provider-azurerm/pull/8719)

**Comment by Satyricon** [comment]
Any update on this one?

**Comment by iharrington** [comment]
Looks like the PR went stale, Any chance this can be added in?

**Comment by aidancopps** [comment]
It looks like this is now supported on the last version of the Azure REST API for file shares so would be great to get support added in Terraform too.

**Comment by arnegrondahl** [comment]
Yes, please add! Would be fantastic.

**Comment by hstringa01** [comment]
This would be great!

**Comment by csorzi** [comment]
FYI this was added with the latest release in #16462

**Comment by rcskosir** [comment]
I am marking this issue closed with #16462, if you believe that is incorrect, please reach out and I can reopen.
